api/src/main/java/org/pmiops/workbench/api/ProfileController.java

```diff
@@ -173,19 +173,36 @@ private DbUser initializeUserIfNeeded() {
     return ResponseEntity.ok(profile);
   }
 
+  private void validateTerraTermsOfServiceStatus(DbUser user) {
+    boolean termsOfServiceAccepted = userService.validateTerraTermsOfService(user);
+    if (!termsOfServiceAccepted) {
+      throw new UnauthorizedException("User has not accepted Terra TOS");
+    }
+  }
+
   @Override
   public ResponseEntity<Profile> getMe() {
-    // Record that the user signed in, and create the user's FireCloud user and free tier billing
+    // Record that the user signed in, confirm that user has accepted Terra TOS
+    // and create the user's FireCloud user and free tier billing
     // project if they haven't been created already.
     // This means they can start using the NIH billing account in FireCloud (without access to
     // the CDR); we will probably need a job that deactivates accounts after some period of
     // not accepting the terms of use.
 
     DbUser dbUser = initializeUserIfNeeded();
+    validateTerraTermsOfServiceStatus(dbUser);
     profileAuditor.fireLoginAction(dbUser);
     return getProfileResponse(dbUser);
   }
 
+  @Override
+  public ResponseEntity<Void> acceptTermsOfService(Integer termsOfServiceVersion) {
+    DbUser loggedInUser = userAuthenticationProvider.get().getUser();
+    userService.submitAouTermsOfService(loggedInUser, termsOfServiceVersion);
+    userService.acceptTerraTermsOfService(loggedInUser);
+    return ResponseEntity.status(HttpStatus.NO_CONTENT).build();
```

Comment:
What does a HttpStatus of NO_CONTENT represent? Does it represent success or failure?

Reply:
yes - this is HTTP 204